plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -100,11 +102,12 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
         `Conflicting write of processing result for ${id} with location key '${locationKey}'`,
       );
     }
+    const sourceEntityRef = stringifyEntityRef(processedEntity);
```

Comment:
@Rugvip As I was testing this implementation with the different usecases I came across that when the PlaceholderProcessor runs the processedEntity will be the actual entity not the location. So if I store the entityRef of that entity in the refreshKeys table I assume the refresh won't happen because in those cases we also should save the location entity. 

I can see something like this happening in the refresh service with the listAncestors method. Which place would be the best suited to find the ancestor location of an entity ?
I think it could happen here, so we only store location entityRefs in the refreshKeys table, or it could happen at the refreshByRefreshKeys method where we would check if the entity ref is a location if not first find its ancestor location and then trigger a refresh.

Reply:
No that's all good actually, for the `PlaceholderProcessor` it is the entity itself that we need to process, not the location. I'd say avoid thinking of the processing in terms of what roles different entities have, and instead what the processors do given different input. If you want something to happen again, then you just want to run the processor again with the same kind of input.

One thing to keep in mind as well is that `readLocation` is just an abstraction around `Location` entities, so essentially `readLocation` is the same thing as processing a `Location` entity.